spacy_wordnet/wordnet_domains.py

```diff
@@ -68,16 +79,43 @@ def wordnet_synsets_for_domain(self, domains: List[str]):
         ]
 
     @staticmethod
-    def __find_synsets(token: Token, lang: str):
+    def __find_synsets(token: Token,
+                       lang: str,
+                       pos: Optional[Union[str, List[str]]] = None) -> List[Synset]:
+        if pos is None:
+            pos = ["verb", "noun", "adj"]
+
+        elif isinstance(pos, str):
+            pos = [pos]
+
+        elif not isinstance(pos, list):
+            try:
+                pos = list(pos)
+
+            except TypeError:
+                raise TypeError("pos argument must be None, type str, or type"
+                                "list.")
+
+        acceptable_pos = {"verb": VERB, "noun": NOUN, "adj": ADJ}
+
+        # check if any element in `pos` is not in `acceptable_pos`
+        if set(pos).difference(acceptable_pos):
+            raise ValueError("pos argument must be a combination of 'verb', "
+                             "'noun', or 'adj'.")
+
+        token_pos: List[int] = [acceptable_pos[k] for k in pos]
         word_variants = [token.text]
-        if token.pos in [VERB, NOUN, ADJ]:
+        if token.pos in token_pos:
             # extend synset coverage using lemmas
             word_variants.append(token.lemma_)
 
         for word in word_variants:
-            token_synsets = wn.synsets(
-                word, pos=spacy2wordnet_pos(token.pos), lang=lang
-            )
+            token_synsets: List[Synset] = []
+            for p in token_pos:
+                token_synsets.extend(wn.synsets(
+                    word, pos=spacy2wordnet_pos(p), lang=lang
```

Comment:
@it176131 Is there any reason to not search by the `token.pos` no `pos` param provided?

Reply:
> @it176131 Is there any reason to not search by the token.pos no pos param provided?

Not sure I understand. Are you saying that I can provide a list as an argument to the `spacy2wordnet_pos` func? Or are you asking why I don't limit strictly to the `token.pos`?

If the latter, I wanted the user to have the ability to filter to multiple `pos`, not just the `token.pos`. As an example, if the `token.pos` is `VERB`, but the user wants `VERB` and `NOUN`, then I'd need provide both options rather than the single.

Of course if the `spacy2wordnet_pos` function can take a list, then I can do away with the loop. But I didn't think it could.

Does that answer the question or am I missing the mark?

Reply:
What I mean is that, if the user does not provide the `pos` parameter, the wordnet search will use all available pos for finding synset.

My proposal is just to keep the old behavior and set as default the `token.pos` value.

I will add a code suggestion to implement my proposal.